src/evaluators/ForStatement.js

```diff
@@ -177,6 +178,7 @@ export default function(
       // 9. If forDcl is an abrupt completion, then
       if (forDcl instanceof AbruptCompletion) {
         // a. Set the running execution context's LexicalEnvironment to oldEnv.
+        realm.onDestroyScope(realm.getRunningContext().lexicalEnvironment);
```

Comment:
If realm.getRunningContext().lexicalEnvironment !== loopEnv, you may need to do more cleanup.